app/src/main/java/com/example/coopertest/LocationActivity.kt

```diff
@@ -0,0 +1,102 @@
+package com.example.coopertest
+
+import android.Manifest
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.location.LocationManager
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.provider.Settings
+import android.view.View
+import androidx.core.app.ActivityCompat
+import kotlinx.android.synthetic.main.activity_location.*
+import androidx.core.app.ComponentActivity
+import androidx.core.app.ComponentActivity.ExtraData
+import androidx.core.content.ContextCompat.getSystemService
+import android.icu.lang.UCharacter.GraphemeClusterBreak.T
+
+
+
+class LocationActivity : AppCompatActivity() {
+
+    private val PERMISSION_ID = 42
+
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_location)
+        if (checkPermissions() && !isLocationEnabled()){
+            ActivateLocation.text= "Activate GPS on your phone"
```

Comment:
Could you move that to strings.xml as well? There is another occurrence below too